lib/codecs/src/decoding/format/json.rs

```diff
@@ -16,7 +18,34 @@ use super::Deserializer;
 
 /// Config used to build a `JsonDeserializer`.
 #[derive(Debug, Clone, Default, Deserialize, Serialize)]
-pub struct JsonDeserializerConfig;
+pub struct JsonDeserializerConfig {
+    #[serde(
+        default,
+        skip_serializing_if = "vector_core::serde::skip_serializing_if_default"
+    )]
+    /// Options for the JSON deserializer.
+    pub json: JsonDeserializerOptions,
+}
+
+/// Options for building a `JsonDeserializer`.
+#[configurable_component]
+#[derive(Debug, Clone, PartialEq, Eq, Derivative)]
+#[derivative(Default)]
+pub struct JsonDeserializerOptions {
+    /// Determines whether or not to replace invalid UTF-8 sequences instead of returning an error.
+    ///
+    /// When true, invalid UTF-8 sequences are replaced with the [`U+FFFD REPLACEMENT CHARACTER`][U+FFFD].
```

Comment:
Should this actually be a link to https://en.wikipedia.org/wiki/Specials_(Unicode_block)#Replacement_character?